### ISSUE TITLE
MultiSelect: titleText prop is undocumented

### ISSUE THREAD

**Issue by jendowns** [carbon-design-system/carbon#6640]
## Detailed description

The `titleText` prop is undocumented (see [`MultiSelect` props list](https://github.com/carbon-design-system/carbon/blob/213cbde854a4958334cff9914b91863ce6debb9d/packages/react/src/components/MultiSelect/MultiSelect.js#L258)) but its very important (without it, [the `MultiSelect` won't pass a11y tests due to missing ARIA attributes](https://github.com/carbon-design-system/carbon/blob/213cbde854a4958334cff9914b91863ce6debb9d/packages/react/src/components/MultiSelect/MultiSelect.js#L179)).

I can propose a PR with some notes on the prop based on my understanding of it. 👍
